internal/gatewayapi/clustersettings.go

```diff
@@ -83,7 +83,13 @@ func translateTrafficFeatures(policy *egv1a1.ClusterSettings) (*ir.TrafficFeatur
 
 func buildTimeout(policy egv1a1.ClusterSettings, r *ir.HTTPRoute) (*ir.Timeout, error) {
 	if policy.Timeout == nil {
-		return nil, nil
+		var ret *ir.Timeout
+		// Some timeout parameters originate from the route. Don't lose them if
+		// there is no timeout section in the policy.
+		if r != nil && r.Traffic != nil {
+			ret = r.Traffic.Timeout.DeepCopy()
+		}
+		return ret, nil
```

Comment:
why are we returning early here, if a route is non empty shouldnt we also compute route timeout set then and then return ?

Reply:
Returning early here because the _policy_ doesn't contain any new timeout information. There's nothing to compute. The only thing that should happen is to not ignore whatever values are already in `ir.Timeout` due to how it was configured from the `HTTPRoute` resource.

Reply:
I think I understand the code flow now, we have two methods
1. [processTimeout ](https://github.com/envoyproxy/gateway/blob/ada279e082485846709ae1250dd2c6b466325626/internal/gatewayapi/route.go#L257)- which builds out the `timeout` struct by processing the HTTPRoute timeout section
2. buildTimeout - which builds out the `timeout` struct by processing BTP

to improve readability, my suggestion would be to rename the functions to `buildRouteTimeout` and `buildBTPTimeout` and add a 3rd func called `mergeTimeouts` which copies over the route timeout fields if set, and gets called in the route section

Reply:
`buildBTPTimeout` is a little misleading. That function is used wherever there are `ClusterSettings` involved - also for things like ext-proc, ext-auth, tracing, and logging cluster definitions.

I'll make a few changes to improve readability though.